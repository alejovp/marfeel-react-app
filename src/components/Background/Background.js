import React from 'react';
import PropTypes from 'prop-types';
import { StyledBackground } from './StyledBackGround';


export const Background = ({ children, ...rest }) => {
    return (
        <StyledBackground {...rest} >
            {children}
        </StyledBackground>
    );
};

Background.propTypes = {
    children: PropTypes.node
};
