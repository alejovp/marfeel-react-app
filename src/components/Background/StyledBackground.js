import styled from 'styled-components';

export const StyledBackground = styled.div`
    height: inherit;
    width: 100%;
    background-color: ${({theme}) => theme.background.color};
`;

StyledBackground.displayName = 'StyledBackground';
