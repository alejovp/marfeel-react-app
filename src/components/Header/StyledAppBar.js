import styled from 'styled-components';

export const StyledAppBar = styled.div`
    position: relative;
    padding: 10px ${({ theme }) => theme.header.padding}px;
    width: 100%;
    height: 60px;
    align-items: center;
    justify-content: center;
    display: flex;
`;

StyledAppBar.displayName = 'StyledAppBar';
