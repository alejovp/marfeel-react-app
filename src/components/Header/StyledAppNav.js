import styled from 'styled-components';

export const StyledAppNav = styled.nav`
    position: relative;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
`;

StyledAppNav.displayName = 'StyledAppNav';
