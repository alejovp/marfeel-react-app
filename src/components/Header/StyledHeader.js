import styled from 'styled-components';

export const StyledHeader = styled.header`
    position: fixed;
    width: 100%;
    top: 0;
    font-size: 1rem;
`;

StyledHeader.displayName = 'StyledHeader';
