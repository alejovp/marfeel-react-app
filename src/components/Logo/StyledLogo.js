import styled from 'styled-components';

export const StyledLogo = styled.a`
    text-decoration: none;
    font-weight: 600;
    cursor: pointer;
    position: relative;
    img {
        height: 36px;
    }
`;
