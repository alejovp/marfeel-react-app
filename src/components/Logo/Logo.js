import React from 'react';
import { StyledLogo } from './StyledLogo';

export const Logo = () => (
    <StyledLogo>
        <img src="https://www.marfeel.com/public/images/landing/header/logo-marfeel.svg" />
    </StyledLogo>
);
