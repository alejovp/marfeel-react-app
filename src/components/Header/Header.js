import React from 'react';
import { Background } from '../Background';
import { Logo } from '../Logo/Logo';
import { StyledAppBar } from './StyledAppBar';
import { StyledAppNav } from './StyledAppNav';
import { StyledHeader } from './StyledHeader';


export const Header = () => (
    <StyledHeader>
        <Background>
            <StyledAppNav>
                <StyledAppBar>
                    <Logo />
                </StyledAppBar>
            </StyledAppNav>
        </Background>
    </StyledHeader>
);
