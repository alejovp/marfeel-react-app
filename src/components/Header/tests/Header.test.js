import React from 'react';
import { Header } from '../Header';
import theme from '../../../../__mocks__/theme.json';
import { shallowWithTheme } from '../../../../scripts/jest/setup';
import { StyledHeader } from '../StyledHeader';
import { Background } from '../../Background';
import { StyledAppNav } from '../StyledAppNav';
import { Logo } from '../../Logo/Logo';


describe('<Header />', () => {

    describe('shape', () => {
        const renderedComponent = shallowWithTheme(<Header />, theme);

        it('will render correctly', () => {
            expect(renderedComponent).toMatchSnapshot();
        });

        it('will render a StyledHeader component', () => {
            expect(renderedComponent.find(StyledHeader)).toHaveLength(1);
        });

        it('will render two Background components', () => {
            expect(renderedComponent.find(Background)).toHaveLength(2);
        });

        it('will render a StyledAppNav component', () => {
            expect(renderedComponent.find(StyledAppNav)).toHaveLength(1);
        });

        it('will render a StyledAppBar component', () => {
            expect(renderedComponent.find(StyledAppNav)).toHaveLength(1);
        });

        it('will render a Logo component', () => {
            expect(renderedComponent.find(Logo)).toHaveLength(1);
        });

    });

});
