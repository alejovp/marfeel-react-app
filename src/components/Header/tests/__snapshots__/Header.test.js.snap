// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> shape will render correctly 1`] = `
<StyledHeader>
  <Background>
    <StyledAppNav>
      <StyledAppBar>
        <Logo />
      </StyledAppBar>
    </StyledAppNav>
  </Background>
</StyledHeader>
`;
